rest-api/src/main/java/gov/cms/qpp/conversion/api/services/CpcFileServiceImpl.java

```diff
@@ -33,15 +36,28 @@
 	}
 
 	/**
-	 * Retrieves the file location id and uses it to retrieve the file
+	 * Retrieves the file location id and retrieves the file if it is an unprocessed cpc+ file
 	 *
 	 * @param fileId {@link Metadata} identifier
 	 * @return file returned as an {@link InputStream}
 	 */
-	public InputStream getFileById(String fileId) {
-		String fileLocationId = dbService.getFileSubmissionLocationId(fileId);
 
-		return storageService.getFileByLocationId(fileLocationId);
+	/**
+	 * Retrieves the file location id and retrieves the file if it is an unprocessed cpc+ file
+	 *
+	 * @param fileId {@link Metadata} identifier
+	 * @return file contents as a {@link String}
+	 * @throws IOException
+	 */
+	public String getFileById(String fileId) throws IOException {
+		String content = "File not found!";
+		Metadata metadata = dbService.getMetadataById(fileId);
+		if (metadata != null && metadata.getCpc() && !metadata.getCpcProcessed()) {
+			content = IOUtils.toString(storageService.getFileByLocationId(metadata.getSubmissionLocator()),
```

Comment:
It would be more appropriate to return a `404` instead of a `200` along with `File not found!`.  An easy way to do this is throw a new exception, and then intercept it in the `ExceptionHandlerControllerV1` class with a new method.